include/boost/math/tools/polynomial.hpp

```diff
@@ -301,7 +301,7 @@ class polynomial
 
    template <class I>
    polynomial(I first, unsigned length)
-      : m_data(first, std::next(first, length))
+      : m_data(first, std::next(first, length + 1))
```

Comment:
Wait . . what happened here?

Reply:
I sized this constructor wrong and it broke the tests; the pointer constructor uses `length + 1`. MSVC needed a constructor for iterators.